pkg/asset/manifests/openshift.go

```diff
@@ -270,6 +271,21 @@ func (o *Openshift) Generate(ctx context.Context, dependencies asset.Parents) er
 			ProvisioningOSDownloadURL: rhcosImage.ControlPlane,
 		}
 		assetData["99_baremetal-provisioning-config.yaml"] = applyTemplateData(baremetalConfig.Files()[0].Data, bmTemplateData)
+
+		// Create certificate to enable TLS for virtual media
+		cert, err := generateTLSCertificate(installConfig.Config.Platform.BareMetal.BootstrapProvisioningIP)
```

Comment:
This all probably works, but I wonder if it's going to be simpler and more self-contained if we do this generation and templating fully inside of our `baremetal/template.go`.

Reply:
GetTemplateData is called from a function that has access to `dependencies`.

Reply:
That makes sense, let me try it out